tests/unit/authenticators/oauth2-password-grant-test.js

```diff
@@ -199,6 +199,26 @@ describe('OAuth2PasswordGrantAuthenticator', () => {
           done();
         });
       });
+
+      describe('when reject with XHR is enabled', () => {
+        beforeEach(() => {
+          authenticator.set('rejectWithXhr', true);
+        });
+
+        it('rejects with xhr object', (done) => {
+          authenticator.authenticate('username', 'password').catch((error) => {
+            expect(error.responseJSON).to.eql({ error: 'invalid_grant' });
+            done();
+          });
+        });
+
+        it('provides access to custom headers', (done) => {
+          authenticator.authenticate('username', 'password').catch((error) => {
+            expect(error.getResponseHeader('X-Custom-Context')).to.eql('foobar');
+            done();
+          });
+        });
```

Comment:
see above for these as well - we should generally not use `done` when returning the promise works as well